pennylane/pauli/dla/structure_constants.py

```diff
@@ -0,0 +1,125 @@
+from typing import List, Union
+from itertools import combinations
+
+import numpy as np
+
+from pennylane.typing import TensorLike
+from pennylane.operation import Operator
+from ..pauli_arithmetic import PauliSentence, PauliWord
+
+
+def _all_commutators(ops):
+    commutators = {}
+    for (j, op1), (k, op2) in combinations(enumerate(ops), r=2):
+        res = op1.commutator(op2)
+        if res != PauliSentence({}):
+            commutators[(j, k)] = res
+
+    return commutators
+
+
+def structure_constants(
+    g: List[Union[Operator, PauliWord, PauliSentence]], pauli: bool = False
+) -> TensorLike:
+    r"""
+    Compute the structure constants that make up the adjoint representation of a Lie algebra.
+
+    Given a DLA :math:`\{iG_1, iG_2, .. iG_d \}` of dimension :math:`d`,
+    the structure constants yield the decomposition of all commutators in terms of DLA elements,
+
+    .. math:: [i G_\alpha, i G_\beta] = \sum_{\gamma = 0}^{d-1} f^\gamma_{\alpha, \beta} iG_\gamma.
+
+    The adjoint representation :math:`\left(\text{ad}(iG_\gamma)\right)_{\alpha, \beta} = f^\gamma_{\alpha, \beta}` is given by those structure constants,
+    which can be computed via
+
+    .. math:: f^\gamma_{\alpha, \beta} = \frac{\text{tr}\left(i G_\gamma \cdot \left[i G_\alpha, i G_\beta \right] \right)}{\text{tr}\left( iG_\gamma iG_\gamma \right)}.
+
+    Note that this is just the projection of the commutator on the DLA element :math:`iG_\gamma` via the trace inner product.
+    The inputs are assumed to be orthogonal. However, we neither assume nor enforce normalization of the DLA elements
+    :math:`G_\alpha`, hence the normalization
+    factor :math:`\text{tr}\left( iG_\gamma iG_\gamma \right)` in the projection.
+
+    Args:
+        g (List[Union[Operator, PauliWord, PauliSentence]]): The (dynamical) Lie algebra for which we want to compute
+            its adjoint representation. DLAs can be generated by a set of generators via :func:`~lie_closure`.
+        pauli (bool): Indicates whether it is assumed that :class:`~PauliSentence` instances are input and returned.
+            This can help with performance to avoid unnecessary conversions from :class:`~PauliSentence` to :class:`~Operator`
+            or :class:`~PauliWord` and vice versa. Note that the input in that case also has to be a list of :class:`~PauliSentence` instances.
+            Default is ``False``.
+
+    Returns:
+        TensorLike: The adjoint representation of shape ``(d, d, d)``.
+
+    **Example**
+
+    Let us generate the DLA of the transverse field Ising model using :func:`~lie_closure`.
+
+    >>> n = 2
+    >>> gens = [PauliSentence({PauliWord({i:"X", i+1:"X"}): 1.}) for i in range(n-1)]
+    >>> gens += [PauliSentence({PauliWord({i:"Z"}): 1.}) for i in range(n)]
+    >>> dla = qml.pauli.lie_closure(gens)
+    >>> print(dla)
+    [1.0 * X(0) @ X(1), 1.0 * Z(0), 1.0 * Z(1), -1.0 * Y(0) @ X(1), -1.0 * X(0) @ Y(1), -1.0 * Y(0) @ Y(1)]
```

Comment:
Very minor, but I wonder if this would look nicer if not using `PauliWord` and `PauliSentence` but instead standard `X` and `Z` ops.

Reply:
Yes good point! This wasnt possible earlier but is now 👍